packages/dx-react-scheduler-demos/src/demo-sources/scheduler-featured-editing/material-ui/demo.jsx

```diff
@@ -0,0 +1,411 @@
+    this.setState({ confirmationVisible: !confirmationVisible });
+  }
+
+  commitDeletedAppointment() {
+    const { data, deletedAppointmentId } = this.state;
+    const nextData = data.filter(appointment => appointment.id !== deletedAppointmentId);
+    this.setState({ data: nextData, deletedAppointmentId: null });
+    this.toggleConfirmationVisible();
+  }
+
+  commitChanges({ added, changed, deleted }) {
+    let { data } = this.state;
+    if (added) {
+      const startingAddedId = data.length > 0 ? data[data.length - 1].id + 1 : 0;
+      data = [
+        ...data,
+        {
+          id: startingAddedId,
+          ...added,
+        },
+      ];
+    }
+    if (changed) {
+      data = data.map(appointment => (
+        changed.id === appointment.id ? { ...appointment, ...changed } : appointment));
+    }
+    if (deleted) {
+      this.setDeletedAppointmentId(deleted.id);
+      this.toggleConfirmationVisible();
+    }
+    this.setState({ data, addedAppointment: {} });
+  }
+
+  render() {
+    const {
+      currentDate,
+      data,
+      confirmationVisible,
+      editingFormVisible,
+    } = this.state;
+    const { classes } = this.props;
+
+    return (
+      <Paper>
+        <Scheduler
+          data={data}
+        >
+          <ViewState
+            currentDate={currentDate}
+          />
+          <EditingState
+            onCommitChanges={this.commitChanges}
+            onEditingAppointmentIdChange={this.onEditingAppointmentIdChange}
+            onAddedAppointmentChange={this.onAddedAppointmentChange}
+          />
+          <DayView
+            startDayHour={9}
+            endDayHour={19}
+          />
+          <WeekView
+            startDayHour={9}
+            endDayHour={19}
+          />
+          <Appointments />
+          <Toolbar />
+          <ViewSwitcher />
+          <AppointmentForm
+            popupComponent={this.appointmentForm}
+            visible={editingFormVisible}
+            onVisibilityChange={this.toggleEditingFormVisible}
+          />
+        </Scheduler>
+
+        <Dialog
+          open={confirmationVisible}
+          onClose={this.cancelDelete}
+        >
+          <DialogTitle>
+            Delete Appointment
+          </DialogTitle>
+          <DialogContent>
+            <DialogContentText>
+              Are you sure you want to delete this appointment?
+            </DialogContentText>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={this.toggleConfirmationVisible} color="primary" variant="outlined">
+              Cancel
+            </Button>
+            <Button onClick={this.commitDeletedAppointment} color="secondary" variant="outlined">
+              Delete
+            </Button>
+          </DialogActions>
+        </Dialog>
+
+        <Button
+          variant="fab"
+          color="secondary"
+          className={classes.addButton}
+          onClick={() => {
+            this.toggleEditingFormVisible();
```

Comment:
This button doesn't toggle a form visibility. It always shows the form. So, we can just set state here:
```
this.setState({ editingFormVisible: true });
```